src/main/java/Deadline.java

```diff
@@ -0,0 +1,49 @@
+import java.time.LocalDateTime;
+import java.time.DateTimeException;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    protected LocalDateTime byDateTime;
+
+    public Deadline(String description, String byDateTime) throws DukeException {
+        super(description);
+        this.taskType = TaskType.DEADLINE;
+        try {
+            this.byDateTime = LocalDateTime.parse(byDateTime, DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm"));
+        } catch (DateTimeParseException e) {
+            throw new DukeException("DateTime Parsing Failed: DateTime Format should follow \"dd/MM/yyyy HH:mm\" " 
+                    + "format." 
+                    + e.getMessage());
+        }
+    }
+
+    public String getByDateTimeString() {
+        try {
+            String string = byDateTime.format(DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm"));
+            return string;
+        } catch (DateTimeException e) {
+            return "DateCannotBeFormatted";
+        }
+    }
+
+    @Override
+    public String toSaveString() {
+        String saveString = super.toSaveString() + "@@@" + this.getByDateTimeString();
+        return saveString;
+    }
+    
+    @Override
+    public String toString() {

```

Comment:
Looks like a clean way to seperate strings and methods to be added into a string.

Reply:
Thanks! :D